Examples/KleinGordon/KleinGordonRHS.hpp

```diff
@@ -0,0 +1,69 @@
+/* GRTeclyn
+ * Copyright 2022 The GRTL collaboration.
+ * Please refer to LICENSE in GRTeclyn's root directory.
+ */
+
+#ifndef KLEINGORDONRHS_HPP_
+#define KLEINGORDONRHS_HPP_
+
+#include "Cell.hpp"
+#include "FourthOrderDerivatives.hpp"
+#include "TensorAlgebra.hpp"
+#include "VarsTools.hpp"
+
+// Problem specific includes
+#include "StateVariables.hpp"
+
+template <class model_t, class deriv_t = FourthOrderDerivatives>
+class KleinGordonRHS
+{
+  public:
+
+    // NOLINTNEXTLINE(bugprone-easily-swappable-parameters)
+    KleinGordonRHS(amrex::Real a_sigma, amrex::Real a_dx, model_t a_model)
+        : m_sigma(a_sigma), m_deriv(a_dx), m_model(a_model) {};
+
+    template <class data_t> struct Vars
+    {
+        data_t phi, Pi;
+
+        template <typename mapping_function_t>
+        AMREX_GPU_DEVICE void enum_mapping(mapping_function_t mapping_function)
+        {
+            VarsTools::define_enum_mapping(mapping_function, c_phi, phi);
+            VarsTools::define_enum_mapping(mapping_function, c_Pi, Pi);
+        }
+    };
+
+    template <class data_t> struct Diff2Vars
+    {
+        data_t phi;
+
+        template <typename mapping_function_t>
+        AMREX_GPU_DEVICE void enum_mapping(mapping_function_t mapping_function)
+        {
+            VarsTools::define_enum_mapping(mapping_function, c_phi, phi);
+        }
+    };
+
+    template <class data_t>
+    AMREX_GPU_DEVICE AMREX_FORCE_INLINE void
+    compute(int i, int j, int k, const amrex::Array4<data_t const> &input,
+            const amrex::Array4<data_t> &output) const;
+
+  protected:
```

Comment:
Nobody is going to derive from this class...
```suggestion
  private:
```